x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/action/TransportDeleteEngineAction.java

```diff
@@ -9,29 +9,30 @@
 
 import org.elasticsearch.action.ActionListener;
 import org.elasticsearch.action.support.ActionFilters;
-import org.elasticsearch.action.support.HandledTransportAction;
 import org.elasticsearch.action.support.master.AcknowledgedResponse;
 import org.elasticsearch.common.inject.Inject;
-import org.elasticsearch.tasks.Task;
+import org.elasticsearch.license.XPackLicenseState;
 import org.elasticsearch.transport.TransportService;
+import org.elasticsearch.xpack.entsearch.EnterpriseSearchTransportAction;
 import org.elasticsearch.xpack.entsearch.engine.EngineIndexService;
 
-public class TransportDeleteEngineAction extends HandledTransportAction<DeleteEngineAction.Request, AcknowledgedResponse> {
+public class TransportDeleteEngineAction extends EnterpriseSearchTransportAction<DeleteEngineAction.Request, AcknowledgedResponse> {
 
     private final EngineIndexService engineIndexService;
 
     @Inject
     public TransportDeleteEngineAction(
         TransportService transportService,
         ActionFilters actionFilters,
-        EngineIndexService engineIndexService
+        EngineIndexService engineIndexService,
+        XPackLicenseState licenseState
     ) {
-        super(DeleteEngineAction.NAME, transportService, actionFilters, DeleteEngineAction.Request::new);
+        super(DeleteEngineAction.NAME, transportService, actionFilters, DeleteEngineAction.Request::new, licenseState);
```

Comment:
If users downgrade their license and want to delete an Engine they're no longer using, will they have a documented way to do this?